NuRadioReco/modules/RNO_G/stationHitFilter.py

```diff
@@ -0,0 +1,525 @@
+        Returns
+        -------
+        self._passed_hit_filter: bool
+            Event passed the Hit Filter or not
+        """
+        self._passed_hit_filter = self._time_checker()
+
+        # If time checker failed or if you want to complete the hit checking regardless
+        if not self._passed_hit_filter or self._complete_hit_check:
+            self._hit_checker()
+
+        # Only update _passed_hit_filter with the hit checker when the time checker did not pass
+        if not self.is_passed_time_checker():
+            self._passed_hit_filter = self.is_passed_hit_checker()
+
+        return self._passed_hit_filter
+
+
+    @register_run()
+    def run(self, evt, station, det=None, noise_RMS_all=None):
+        """
+        Runs the Hit Filter.
+
+        Parameters
+        ----------
+        evt: `NuRadioReco.framework.event.Event`
+            Using the event object to get the trigger type
+        station: `NuRadioReco.framework.station.Station`
+            The station to use the Hit Filter on
+        det: Detector object | None
+            Detector object (not used in this method,
+            included to have the same signature as other NuRadio classes)
+        noise_RMS_all: 1-D numpy array (default: None)
+            Noise RMS values of all 24 channels, if not given the Hit Filter will calculate them for the 15 in-ice channels
+
+        Returns
+        -------
+        self.is_passed_hit_filter(): bool
+            Event passed the Hit Filter or not
+        """
+        # This implicitly obeys the channel mapping
+        traces = np.array([np.array(channel.get_trace()) for channel in station.iter_channels() if channel.get_id() in self._in_ice_channels])
+        times = np.array([np.array(channel.get_times()) for channel in station.iter_channels() if channel.get_id() in self._in_ice_channels])
+
+        if noise_RMS_all is not None:
+            noise_RMS = noise_RMS_all[self._in_ice_channels]  # HACK: use channel IDs to index noise_RMS
+        else:
+            noise_RMS = noise_RMS_all
+
+        self.set_up(traces, times, noise_RMS)
+        self.apply_hit_filter()
+
+        trigger_type = evt.get_station().get_first_trigger().get_name()
+
+        if trigger_type == "FORCE":
+            self.__count_FT += 1
+            self.__count_passed_FT += int(self.is_passed_hit_filter())
+            self.__is_wanted_trigger_type = False
+        elif "RADIANT" in trigger_type:
+            self.__count_RADIANT += 1
+            self.__count_passed_RADIANT += int(self.is_passed_hit_filter())
+            self.__is_wanted_trigger_type = False
+        elif trigger_type == "UNKNOWN":
+            self.__count_UNKNOWN += 1
+            self.__count_passed_UNKNOWN += int(self.is_passed_hit_filter())
+            self.__is_wanted_trigger_type = True
+        else:
+            self.__count_RF += 1
+            self.__count_passed_RF += int(self.is_passed_hit_filter())
+            self.__is_wanted_trigger_type = True
```

Comment:
I thought an `"UNKNOWN"` trigger is always an RADIANT trigger @shallmann  or @cozzyd can you confirm? If so you can add in l290 `self.__is_wanted_trigger_type = False` and only set it to true if trigger same is `"LT"`

Reply:
I think in principle the day can also produce "PPS" trigger although I have never seen them. I would make an explicit `trigger_type == "LT"` and have an `else` which deals with `"UNKNOWN"` and everything else.

Reply:
I took the liberty of changing it